x-pack/plugins/apm/common/apm_settings.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { mapValues } from 'lodash';
+
+export enum ApmSettingName {
+  AgentConfigurationAvailable = 'agentConfigurationAvailable',
+  ConfigurableIndicesAvailable = 'configurableIndicesAvailable',
+  InfrastructureTabAvailable = 'infrastructureTabAvailable',
+  InfraUiAvailable = 'infraUiAvailable',
+  MigrationToFleetAvailable = 'migrationToFleetAvailable',
+  SourcemapUploadAvailable = 'sourcemapUploadAvailable',
+  SpacesAvailable = 'spacesAvailable',
+}
+
+const apmSettingsMap = {
+  [ApmSettingName.AgentConfigurationAvailable]: {
+    default: true,
+    type: t.boolean,
+  },
+  [ApmSettingName.ConfigurableIndicesAvailable]: {
+    default: true,
+    type: t.boolean,
+  },
+  [ApmSettingName.InfrastructureTabAvailable]: {
+    default: true,
+    type: t.boolean,
+  },
+  [ApmSettingName.InfraUiAvailable]: {
+    default: true,
+    type: t.boolean,
+  },
+  [ApmSettingName.MigrationToFleetAvailable]: {
+    default: true,
+    type: t.boolean,
+  },
+  [ApmSettingName.SourcemapUploadAvailable]: {
```

Comment:
We talked about also disabling the list endpoint and  delete endpoint. In that case, what about having a single setting for all of them?

```suggestion
  [ApmSettingName.SourcemapApiAvailable]: {
```

Reply:
That's fine with me - is that confirmed?

Reply:
It's up to us to decide. Unless you have any objections I suggest that we disable all the source map APIs.